types/src/deploy/id.rs

```diff
@@ -1,39 +1,32 @@
-use std::fmt::{self, Debug, Display, Formatter};
+use alloc::vec::Vec;
+use core::fmt::{self, Display, Formatter};
 
+#[cfg(feature = "datasize")]
 use datasize::DataSize;
 use serde::{Deserialize, Serialize};
 
-use casper_types::bytesrepr::{self, FromBytes, ToBytes};
-#[cfg(any(feature = "testing", test))]
-use casper_types::testing::TestRng;
-
+#[cfg(doc)]
+use super::Deploy;
 use super::{ApprovalsHash, DeployHash};
+use crate::bytesrepr::{self, FromBytes, ToBytes};
+#[cfg(any(feature = "testing", test))]
+use crate::testing::TestRng;
 
-/// The unique identifier of a deploy, comprising its [`DeployHash`] and [`ApprovalsHash`].
+/// The unique identifier of a [`Deploy`], comprising its [`DeployHash`] and [`ApprovalsHash`].
 #[derive(
-    Copy,
-    Clone,
-    DataSize,
-    Ord,
-    PartialOrd,
-    Eq,
-    PartialEq,
-    Hash,
-    Serialize,
-    Deserialize,
-    Debug,
-    Default,
+    Copy, Clone, Ord, PartialOrd, Eq, PartialEq, Hash, Serialize, Deserialize, Debug, Default,
 )]
+#[cfg_attr(feature = "datasize", derive(DataSize))]
 #[serde(deny_unknown_fields)]
-pub struct Id {
+pub struct DeployId {
```

Comment:
I like it. Should we rename the file to match (deploy_id.rs).

Reply:
Done in [2fa6f99](https://github.com/casper-network/casper-node/pull/4053/commits/2fa6f99034f43c4d559f9bcc118c18144bb82129) (along with a few similar others).